app/graphql/schema/queries/customer-query-response.js

```diff
@@ -0,0 +1,49 @@
+const cosmos = require('../../../cosmos')
+const { cosmosConfig } = require('../../../config')
+const { convertCosmosTimestamp } = require('../../../utils')
+
+const customerQueryResponse = async (_root, args, context) => {
+  try {
+    const { queriesDatabase } = await cosmos()
+
+    const querySpec = {
+      query: 'SELECT * FROM customerQueryResponse cq WHERE cq.id = @id',
+      parameters: [{ name: '@id', value: `${args.id}` }]
+    }
+
+    const response = await queriesDatabase
+      .container(cosmosConfig.queriesContainer)
+      .items.query(querySpec)
+      .fetchAll()
+
+    if (!args.id) {
+      throw new Error('id must be provided')
+    }
+
+    if (!response.resources.length) {
+      throw new Error(`No customer query data found for id ${args.id}`)
+    }
+
+    const resource = response.resources[0]
+    const ukTimestamp = resource ? convertCosmosTimestamp(resource._ts) : null
+
+    return {
+      code: 200,
```

Comment:
Can these return statements be split so we have the resource (the actual data we're getting back) separate from the request details eg the success/message/code

```suggestion
return {
    code: response.statusCode,
    success,
    message,
    resource: { 
          ...response.resource
           timestamp: convertCosmosTimestamp(response.resource._ts)
     }
  }
  ```

Reply:
Not sure if I'm doing something wrong, but looks like when it comes to the queries and fetching the data, it's not fetching a `statusCode`? Have tried 2 methods so far:

### **1. Splitting the resource from the request details:**
```
    const resource = response.resources[0]
    const timestamp = convertCosmosTimestamp(resource._ts)
    const success = response.statusCode >= 200 && response.statusCode < 300
    const message = success ? 'Query to Cosmos DB has been successful' : 'Unable to retrieve data from Cosmos DB'

    return {
      code: resource.statusCode,
      success,
      message,
      resource: {
        ...response.resources[0],
        timestamp
      }
    }
```
With the above everything comes back as `null` except for `success` and `message` (output from Postman copied in below):
```
{
    "data": {
        "customerQueryResponse": {
            "code": null,
            "success": false,
            "message": "Unable to retrieve data from Cosmos DB",
            "id": null,
            "ticketId": null,
            "timestamp": null,
            "internalUser": null,
            "name": null,
            "heading": null,
            "body": null
        }
    }
}
```
I'm thinking the reason for this is because the `typeDef` is set up with the resource and request details altogether:
```
type CustomerQueryResponse {
  code: Int
  success: Boolean
  message: String
  id: String
  ticketId: String
  timestamp: String
  internalUser: Boolean
  name: String
  heading: String
  body: String
}
```
And because of this I'm wondering if it's not possible to contain the `customerQueryResponse` within the `resource` in the return object?

### **2. Not splitting the resource from the request details:**
```
    const resource = response.resources[0]
    const timestamp = convertCosmosTimestamp(resource._ts)
    const success = response.statusCode >= 200 && response.statusCode < 300
    const message = success ? 'Query to Cosmos DB has been successful' : 'Unable to retrieve data from Cosmos DB'

    return {
      code: resource.statusCode,
      success,
      message,
      ...response.resources[0],
      timestamp
    }
```
With the above I'm able to get all the details returned (have copied in the output from Postman below), but again the `statusCode` still seems inaccessible? I tried logging the `statusCode` but it comes back as `undefined` (not sure why that's the case).

Output:
```
{
    "data": {
        "customerQueryResponse": {
            "code": null,
            "success": false,
            "message": "Unable to retrieve data from Cosmos DB",
            "id": "6c739032-8648-4a23-9bae-01ef50c7f321",
            "ticketId": "36e793f4-c32c-436c-8932-f369f1177844",
            "timestamp": "25/06/2024, 17:26:56",
            "internalUser": true,
            "name": "fawegr",
            "heading": "fergs",
            "body": "acv"
        }
    }
}
```
For now, I've pushed the latest changes following your suggestion to split the resource from the request details, was wondering if you could take a look and see if I'm missing something? This is also a _very_ long comment so happy to jump on a call if that's quicker?